packages/docusaurus-plugin-content-blog/src/plugin-content-blog.d.ts

```diff
@@ -418,7 +418,10 @@ yarn workspace v1.22.19image` is a collocated image path, this entry will be the
     /** A callback to customize the reading time number displayed. */
     readingTime: ReadingTimeFunctionOption;
     /** Governs the direction of blog post sorting. */
-    sortPosts: 'ascending' | 'descending';
+    sortPosts:
+      | 'ascending'
+      | 'descending'
+      | ((a: BlogPost, b: BlogPost) => number);
```

Comment:
I'd prefer if the sortPosts took the unsorted array and did the sort.

`({blogPosts: BlogPost[]}) => undefined | BlogPost[];`

If user returns undefined we assume they sorted the provided array by mutating it directly

If user returns an array we assume they sorted the array by creating an updated copy of it